src/main/resources/com/owlcyberdefense/gpssps/gpssps/xsd/gpssps.dfdl.xsd

```diff
@@ -0,0 +1,366 @@
+                <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+              </xs:appinfo>
+            </xs:annotation>
+            <xs:group ref="gpssps:subframe1Word4"/>
+            <xs:sequence>
+              <xs:annotation>
+                <xs:appinfo source="http://www.ogf.org/dfdl/">
+                  <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                  <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                </xs:appinfo>
+              </xs:annotation>
+              <xs:group ref="gpssps:subframe1Word5"/>
+              <xs:sequence>
+                <xs:annotation>
+                  <xs:appinfo source="http://www.ogf.org/dfdl/">
+                    <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                    <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                  </xs:appinfo>
+                </xs:annotation>
+                <xs:group ref="gpssps:subframe1Word6"/>
+                <xs:sequence>
+                  <xs:annotation>
+                    <xs:appinfo source="http://www.ogf.org/dfdl/">
+                      <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                      <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                    </xs:appinfo>
+                  </xs:annotation>
+                  <xs:group ref="gpssps:subframe1Word7"/>
+                  <xs:sequence>
+                    <xs:annotation>
+                      <xs:appinfo source="http://www.ogf.org/dfdl/">
+                        <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                        <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                      </xs:appinfo>
+                    </xs:annotation>
+                    <xs:group ref="gpssps:subframe1Word8"/>
+                    <xs:sequence>
+                      <xs:annotation>
+                        <xs:appinfo source="http://www.ogf.org/dfdl/">
+                          <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                          <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                        </xs:appinfo>
+                      </xs:annotation>
+                      <xs:group ref="gpssps:subframe1Word9"/>
+                      <xs:sequence>
+                        <xs:annotation>
+                          <xs:appinfo source="http://www.ogf.org/dfdl/">
+                            <dfdl:newVariableInstance ref="gpssps:priorD25ToD30" defaultValue="$gpssps:nextD25ToD30"/>
+                            <dfdl:newVariableInstance ref="gpssps:nextD25ToD30"/>
+                          </xs:appinfo>
+                        </xs:annotation>
+                        <xs:group ref="gpssps:subframe1Word10"/>
```

Comment:
My one concern here is that this doesn't really scale that well due to this cascading of the priorD25toD30 stuff. I imagine once you add all the subframe's up to 5 that this schema will be unreadably deep. 

Is the issue that the parity calculation depends on all previous words, but each word stores the current parity value? So you kindof need a way for each word to use the current parity value and also read it after the word is complete?

I wonder if rather than using variables, maybe we could have a single pairty layer that constantly calculates the value, and that value can be queried with a function/UDF? So something like this:

```xml
...
<xs:element name="parityHOW" type="gpssps:bits2"
  dfdl:outputValueCalc="{ gpssps:currentParityValue() }">
  <xs:annotation>
    <xs:appinfo source="http://www.ogf.org/dfdl/">
      <dfdl:assert test='{ . eq gpssps:currentParityValue() }'/>
    </xs:appinfo>
  </xs:annotation>
</xs:element>
...
```
I'm not sure if this would even work, since it requires that the parity layer to not read ahead and at all? And I can imagine suspensions during unparse make this even more complicated?

Reply:
Deeply nested, but it is straight-line code, not a branch nest. Maybe that's good for something. 

Alternatively, these messages consist of this pile of different message words, each with bits D25 to D30. There's no reason to have only one such variable and keep creating new instances of it. One could have a separate variable for the bits at the end of each word. That makes the nesting shallower, at the expense of a bunch of variables getting new instances at the start of a message.  I will take a look and consider if that improves things or not. 

The reason for this variables and layers intensive approach is to avoid interaction with the unparser and suspension logic. In principle this scheme doesn't interact with that at all. If the layer transform logic is keeping state then they would interact significantly. That doesn't mean we can't entertain that idea. It just requires additional consideration to think through the implications.

Reply:
> Deeply nested, but it is straight-line code, not a branch nest. Maybe that's good for something.

Keep in mind, even though it is straight line, I think the nested sequences are going to cause SequenceCombinators to be built, so the call stack will still be nested. If we get too deep (like maybe once all the other frames are added), potentially it gets into an issue with the stack size?

> Deeply nested, but it is straight-line code, not a branch nest. Maybe that's good for something.

I guess this is maybe a valid use case for being able to specify a variable name for a layer?

> That makes the nesting shallower, at the expense of a bunch of variables getting new instances at the start of a message

Maybe this would be a good reason to support variable arrays, even if we only support fixed length arrays? Without thinking about this too much, I'm imagining something like this:

```xml
<dfdl:defineVariable name="parities" type="gpssps:bits6" count="30" />

<xs:sequence dfdl:layerTransform="gpssps:parity" dfdl:transformProperty="prev=parities[1] res=parities[2]">
  ...
</xs:sequence>
<xs:sequence dfdl:layerTransform="gpssps:parity" dfdl:transformProperty="prev=parities[2] res=parities[3]">
  ...
</xs:sequence>
```
This avoids needing to define a ton of variables, but I would imagine would be functionally the same? We would still have the same limitations like you can only write to an index once, once an index is read you can't write to it, etc. So it really just becomes short hand for creating a bunch of variables. So maybe it's not that useful, just slightly less verbose...